Doc/library/types.rst

```diff
@@ -86,6 +86,28 @@ Dynamic Type Creation
 
    :pep:`560` - Core support for typing module and generic types
 
+.. function:: get_original_bases(cls, /)
+
+    Return the objects in the bases list in the class's definition before
+    they could have been modified by ``__mro_entries__``. This is useful for
+    introspecting ``Generic``\s.
+
+    Examples::
+
+        from typing import TypeVar, Generic
+
+        T = TypeVar("T")
+
+        class Foo(Generic[T]): ...
+
+        class Bar(Foo[int], float): ...
+
+        get_original_bases(Foo) == (Generic[T],)
+        get_original_bases(Bar) == (Foo[int], float)
```

Comment:
Add examples of directly accessing `__bases__`, to more clearly illustrate the difference in what this new function provides

```suggestion
        Foo.__bases__ == (Generic,)
        get_original_bases(Foo) == (Generic[T],)

        Bar.__bases__ == (Foo, float)
        get_original_bases(Bar) == (Foo[int], float)

```